commcare_connect/opportunity/tasks.py

```diff
@@ -333,17 +332,18 @@ def bulk_approve_completed_work():
         for completed_work in completed_works:
             if completed_work.completed_count > 0:
                 approved_count = completed_work.approved_count
-                if access.opportunity.managed:
-                    visits = completed_work.uservisit_set.filter(review_status=VisitReviewStatus.agree).values_list(
-                        "status", "reason"
-                    )
-                else:
-                    visits = completed_work.uservisit_set.values_list("status", "reason")
-                if any(status == "rejected" for status, _ in visits):
+                visits = completed_work.uservisit_set.values_list("status", "reason", "review_status")
+                if any(status == "rejected" for status, *_ in visits):
                     completed_work.update_status(CompletedWorkStatus.rejected)
-                    completed_work.reason = "\n".join(reason for _, reason in visits if reason)
-                elif all(status == "approved" for status, _ in visits):
+                    completed_work.reason = "\n".join(reason for _, reason, _ in visits if reason)
+                elif all(status == "approved" for status, *_ in visits):
                     completed_work.update_status(CompletedWorkStatus.approved)
+                if (
+                    access.opportunity.managed
+                    and not all(review_status == "agree" for *_, review_status in visits)
```

Comment:
does this get set automatically somewhere for autoapproved forms?

Reply:
Added. [4deb642](https://github.com/dimagi/commcare-connect/pull/383/commits/4deb642b2950301179b299d1ed8e227d08d72fac)